File: app/src/main/java/com/ardev/idroid/ui/home/ProjectProvider.java
package com.ardev.idroid.ui.home;

import android.content.Context;

import androidx.annotation.NonNull;
import java.io.File;
import com.ardev.builder.project.Project;


public class ProjectProvider {

    private static Project project;

    public static void init(@NonNull Project project) {
        this.project = project;
    }

    public static Project getProject() {
        if (project == null) {
            throw new IllegalStateException("init() has not been called yet in ProjectManager.class ");
        }
        return project;

    }
}

File: app/src/main/java/com/ardev/idroid/ui/projects/ProjectProvider.kt
package com.ardev.idroid.ui.projects

import android.content.Context
import java.io.File
import java.lang.ref.WeakReference
import com.ardev.builder.project.Project
import com.ardev.idroid.common.ext.*

object ProjectProvider {
    private lateinit var ref: WeakReference<Project>

    fun init(project: Project) {
        ref = WeakReference(project)
    }

    fun getProject(): Project {
        if (ref.get() == null) restartApp() 
        return ref.get()!!
    }
}
